File: app/src/main/kotlin/jb/openware/app/tools/UserConfig.java
package in.afi.codekosh.tools;

import android.content.Context;
import android.content.SharedPreferences;

public class UserConfig {
    private static final String PREF_NAME = "user_config";
    private static final String KEY_NAME = "name";
    private static final String KEY_EMAIL = "email";
    private static final String KEY_BADGE = "badge";
    private static final String KEY_PASSWORD = "password";
    private static final String KEY_PROFILE_URL = "profileUrl";
    private static final String KEY_LOGGED_IN = "loggedIn";
    private static final String KEY_UID = "uid";
    private final SharedPreferences preferences;
    private final SharedPreferences.Editor editor;

    public UserConfig(Context context) {
        preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        editor = preferences.edit();
    }

    public int getBadge() {
        return preferences.getInt(KEY_BADGE, 0);
    }

    public void setBadge(int n) {
        preferences.edit().putInt(KEY_BADGE, n).apply();
    }

    public String getEmail() {
        return preferences.getString(KEY_EMAIL, "");
    }

    public String getName() {
        return preferences.getString(KEY_NAME, "");
    }

    public void setName(String name) {
        editor.putString(KEY_NAME, name).apply();
    }

    public String getPassword() {
        return preferences.getString(KEY_PASSWORD, "");
    }

    public String getProfileUrl() {
        return preferences.getString(KEY_PROFILE_URL, "");
    }

    public void setProfileUrl(String string2) {
        preferences.edit().putString(KEY_PROFILE_URL, string2).apply();
    }

    public String getUid() {
        return preferences.getString(KEY_UID, "");
    }

    public boolean isLoggedIn() {
        return preferences.getBoolean(KEY_LOGGED_IN, false);
    }

    public void saveLoginDetails(String email, String password, String uid) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(KEY_LOGGED_IN, true);
        editor.putString(KEY_EMAIL, email);
        editor.putString(KEY_PASSWORD, password);
        editor.putString(KEY_UID, uid);
        editor.apply();
    }

    public void logout() {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(KEY_LOGGED_IN, false);
        editor.putString(KEY_EMAIL, null);
        editor.putString(KEY_PASSWORD, null);
        editor.putString(KEY_PROFILE_URL, null);
        editor.putString(KEY_UID, null);
        editor.putInt(KEY_BADGE, 0);
        editor.apply();

    }

}

File: app/src/main/kotlin/jb/openware/app/util/UserConfig.kt
package jb.openware.app.util

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit

class UserConfig(context: Context) {

    private val prefs: SharedPreferences =
        context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)

    var badge: Int
        get() = prefs.getInt(KEY_BADGE, 0)
        set(value) {
            prefs.edit { putInt(KEY_BADGE, value) }
        }

    var email: String
        get() = prefs.getString(KEY_EMAIL, "") ?: ""
        private set(value) {
            prefs.edit { putString(KEY_EMAIL, value) }
        }

    var name: String
        get() = prefs.getString(KEY_NAME, "") ?: ""
        set(value) {
            prefs.edit { putString(KEY_NAME, value) }
        }

    var password: String
        get() = prefs.getString(KEY_PASSWORD, "") ?: ""
        private set(value) {
            prefs.edit { putString(KEY_PASSWORD, value) }
        }

    var profileUrl: String
        get() = prefs.getString(KEY_PROFILE_URL, "") ?: ""
        set(value) {
            prefs.edit { putString(KEY_PROFILE_URL, value) }
        }

    var uid: String
        get() = prefs.getString(KEY_UID, "") ?: ""
        private set(value) {
            prefs.edit { putString(KEY_UID, value) }
        }

    val isLoggedIn: Boolean
        get() = prefs.getBoolean(KEY_LOGGED_IN, false)

    fun saveLoginDetails(email: String, password: String, uid: String) {
        prefs.edit {
            putBoolean(KEY_LOGGED_IN, true)
            putString(KEY_EMAIL, email)
            putString(KEY_PASSWORD, password)
            putString(KEY_UID, uid)
        }
    }

    fun logout() {
        prefs.edit {
            putBoolean(KEY_LOGGED_IN, false)
            remove(KEY_EMAIL)
            remove(KEY_PASSWORD)
            remove(KEY_PROFILE_URL)
            remove(KEY_UID)
            putInt(KEY_BADGE, 0)
        }
    }

    companion object {
        private const val PREF_NAME = "user_config"
        private const val KEY_NAME = "name"
        private const val KEY_EMAIL = "email"
        private const val KEY_BADGE = "badge"
        private const val KEY_PASSWORD = "password"
        private const val KEY_PROFILE_URL = "profileUrl"
        private const val KEY_LOGGED_IN = "loggedIn"
        private const val KEY_UID = "uid"
    }
}